app/plugin_api.go

```diff
@@ -625,6 +626,10 @@ func (api *PluginAPI) SendMail(to, subject, htmlBody string) *model.AppError {
 
 // Plugin Section
 
+func (api *PluginAPI) UploadPlugin(file io.Reader, replace bool) (*model.Manifest, *model.AppError) {
+	return api.app.InstallPlugin(file, replace)
```

Comment:
The API appears that way inconsistent to me - If I install an application on my OS, it is also not directly started. I'd prefer to stick to the same naming convention.

Reply:
> If I install an application on my OS, it is also not directly started.

Well, it depends on the application. For applications with an UI this is true in most cases. But applications, that run in the background, are mostly started after the installation, e.g. `sudo apt install postgresql`.

Please also note that this is consistent with the REST API method. See https://api.mattermost.com/#tag/plugins%2Fpaths%2F~1plugins%2Fpost

Reply:
Even with applications running in the background it's not 100% consistent, there are various services that require explicit start (e.g. Graylog) - If we want to change the naming conventions I'd like to do it in a consistent fashion, personally I don't mind calling it "Upload" because that is clear to only refer to the upload, and not starting.

The REST API needs to be adapted then as well.